### ISSUE TITLE
RadioButton: label is not announced with a helper or error message

### ISSUE THREAD

**Issue by silvalaura** [cengage/react-magma#1928]
**Description of bug:**
Radio Button text label is not announced with helper text or error message
 
<img width="891" height="525" alt="image" src="https://github.com/user-attachments/assets/6a6aea14-4843-4fc7-8fb8-9fe357696f7d" />

<img width="738" height="293" alt="image" src="https://github.com/user-attachments/assets/88ad4b2d-0f7e-4ea5-bc18-e4836f34de33" />

<img width="1121" height="422" alt="image" src="https://github.com/user-attachments/assets/2788dcba-2d02-47bf-b1a6-51f414726f2f" />

**Devices:**
- MacOS Sonoma + Firefox/Chrome 🚫 
- MacOS Sequoia + Firefox/Chrome/Safari 🚫 
- Windows 10/11 + Firefox/Chrome 🚫 
- MacOS Sonoma + Safari ✅ 

**Steps to reproduce:**
1. Open MacOS/Windows
2. Turn on Screenreader
3. Radio Button with helper/error message
4. Focus on it using a tab
5. Check that a button (label) is not announced

**Expected Behavior**
For all browsers and screen reader combinations, the screen reader should announce the text associated with the radio button
